### PR TITLE
:fire: require python >= 3.7

### PR DESCRIPTION
**Pull request by kba**
In the Dockerfile, install python3.7 from the deadsnakes PPA instead of the 18.04 standard python3.6 and stop testing for python3.6.

With the updated base image, we can then tackle the update in ocrd_all.

### CODE REVIEW

**Comment by bertsky** [comment]
GH already tells me that on their side (GH Actions), the ubuntu:18.04 base image is deprecated now. I don't know about Circle CI. 

But we do have the problem that most of our `deps-ubuntu` rules (in various places) do not differentiate between e.g. `python3-dev` and (say) `python3.7-dev`. Why should they? Because Ubuntu 18's default is actually python-3.6-dev, which is wrong.

So I guess I am saying we should **also** upgrade the base image now. Surprising myself there, but my guess is the other way is more painful overall.

**Review by bertsky — APPROVED** [review]

**Comment by kba** [comment]
> So I guess I am saying we should also upgrade the base image now. Surprising myself there, but my guess is the other way is more painful overall.

Agreed, let's make the jump now. I'll adapt.

File: Dockerfile [hunk 1]
```diff
@@ -19,21 +19,22 @@ COPY README.md .
...
 RUN apt-get update && apt-get -y install software-properties-common \
-    && add-apt-repository ppa:deadsnakes/ppa \
     && apt-get update && apt-get -y install \
         ca-certificates \
-        python3.7-dev \
-        python3.7-venv \
+        python3-dev \
+        python3-pip \
+        python3-venv \
+        gcc \
```
Comment (bertsky) [inline Dockerfile:27]: Oh? Why is this necessary in core now? (AFAICS we have it ocrd_all/Makefile's `deps-ubuntu` → `deps-ubuntu-modules` → `CUSTOM_DEPS`.)

**Review by bertsky — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (kba) [reply Dockerfile:27]: `gcc` and `python3-dev` because we need to install shapely from source in core itself, otherwise CI would fail at `make install`, wouldn't it? `python3-venv` and `python3-pip` are probably not necessary, I'll check.

Comment (bertsky) [reply Dockerfile:27]: right, of course! I forgot that core needs to be self-serving. (Should perhaps get a comment though so we can remove again when not needed.)

File: .circleci/config.yml [hunk 2]
```diff
@@ -80,8 +80,8 @@ workflows:
               only: master
   test-pull-request:
     jobs:
-      - test-python36
       - test-python37
       - test-python38
       - test-python39
+      - test-python310
       - test-macos
```
Comment (stweil) [inline .circleci/config.yml:87]: Should Python 3.11 be added, too?

File: .circleci/config.yml [hunk 3]
```diff
@@ -16,39 +16,39 @@ jobs:
       - run: make deps-test install PIP=pip3
       - run: make test benchmark PYTHON=python3
 
-  test-python36:
+  test-python37:
     docker:
-      - image: python:3.6.12
+      - image: python:3.7.9
```
[the remark this answers is not in the document]
Comment (kba) [reply .circleci/config.yml]: Sure, c35a753b8

File: .circleci/config.yml (hunk 2 above)
Comment (kba) [reply .circleci/config.yml:98]: Yes 8698fc2ff
